src/axom/primal/utils/ZipPoint.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_PRIMAL_ZIP_POINT_HPP_
+#define AXOM_PRIMAL_ZIP_POINT_HPP_
+
+#include "axom/config.hpp"
+#include "axom/core/StackArray.hpp"
+#include "axom/slic/interface/slic.hpp"
+
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/utils/ZipIndexable.hpp"
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+/*!
+ * \brief Implements ZipIndexable for a primal::Point instantiation
+ */
+template <typename FloatType, int NDIMS>
+struct ZipBase<Point<FloatType, NDIMS>>
+{
+  using GeomType = Point<FloatType, NDIMS>;
+  using CoordType = FloatType;
```

Comment:
Minor nit: is there a reason to not make the template arg 'CoordType' to simplify?

Reply:
I think this was originally to get around some of the primal geometry primitives not defining a `CoordType` trait. I'll go ahead and remove it, now that `CoordType` is at least defined in the geometry primitives we're specializing `ZipBase` for.